ipcore/vhdl_node.py

```diff
@@ -0,0 +1,373 @@
+        else:
+            right_hand_side = self._get_right_hand_side_vector_value(new_signal_value)
+        return f"{self.name} <= {right_hand_side};\n"
+    def _get_right_hand_side_vector_value(self, new_signal_value):
+        new_data_type = new_signal_value.underlying_data_type
+        if self.underlying_data_type is None or new_data_type is None:
+            if self.length == new_signal_value.length:
+                right_hand_side = new_signal_value.name
+            elif self.underlying_data_type == new_data_type:
+                # Both have no underlying_data_type, so just resize it
+                right_hand_side = f"std_logic_vector(resize(unsigned({new_signal_value.name}), {self.length}))"
+            elif self.length > new_signal_value.length and new_data_type is not None:
+                # If extending signal, don't worry that assigment is to signal with no underlying data type
+                conversion = "signed" if new_data_type.signed else "unsigned"
+                right_hand_side = f"std_logic_vector(resize({conversion}({new_signal_value.name}), {self.length}))"
+            elif self.underlying_data_type is not None:
+                conversion = "signed" if self.underlying_data_type.signed else "unsigned"
+                right_hand_side = f"std_logic_vector(resize({conversion}({new_signal_value.name}), {self.length}))"
+            else:
+                raise ValueError("Assigning signals of different length requires they either both or neither have underlying data types.\n"
+                                 f"Attemped assigning {new_signal_value.name} ({new_signal_value.length}) to {self.name} ({self.length})")
+        elif self.underlying_data_type == new_data_type:
+            right_hand_side = new_signal_value.name
+        else:
+            if new_signal_value.data_type == "std_logic_vector":
+                conversion = "signed" if new_data_type.signed else "unsigned"
+                input_value = f"{conversion}({new_signal_value.name})"
+            else:
+                input_value = new_signal_value.name
+            fixed_inputs = f"{input_value}, {new_signal_value.length}, {new_data_type.frac_len}, " \
+            + f"{self.length}, {self.underlying_data_type.frac_len}"
+
+            right_hand_side = f"resize_fixed({fixed_inputs})"
+            if self.data_type == "std_logic_vector":
+                right_hand_side = f"std_logic_vector({right_hand_side})"
+        return right_hand_side
+    def _get_default_str(self):
+        if (not isinstance(self.default_value, str)) and self.underlying_data_type is not None:
+            word_len = self.underlying_data_type.word_len
+            frac_len = self.underlying_data_type.frac_len
+            if word_len == 1:
+                default_bit_string = f"'{self.default_value}'"
+            else:
+                default_bit_string = num_to_bitstring(self.default_value, word_len, frac_len)
```

Comment:
Same comment as before about handling the `word_len == 1` case inside of `num_to_bitstring`